db/migrate/20160427150421_add_project_state.rb

```diff
@@ -0,0 +1,6 @@
+class AddProjectState < ActiveRecord::Migration
+  def change
+    add_column :projects, :state, :integer
+    add_index :projects, :state
```

Comment:
I'm wondering if this should be a sparse index? I.e. with a predicate to only index set values and not all:
`add_index :projects, :state, where: "(state IS NOT NULL)", algorithm: :concurrently`